### ISSUE TITLE
Swoole coroutine $redis->get() 2x slower

### ISSUE THREAD

**Issue by jobs-git** [swoole/swoole-src#2253]
Application throughput reduces to half when I incorporate `$redis->get('key')`. `redis` is just retrieving single char `A`.

Here is the application structure

```php
$pool_redis = new \SplQueue;

$server = new swoole_http_server("127.0.0.1", 9501, SWOOLE_BASE);
$server->set(['worker_num' => 1]);

$server->on('request', function (swoole_http_request $req, swoole_http_response $response) use ($pool_redis) {
    if ($pool_redis->isEmpty()) {
        $redis = new swoole_redis;
        $redis->connect('127.0.0.1', 6379);
        $redis->set('key', 'A');
    } else {
        $redis = $pool_redis->dequeue();
    }
    //with this I get half throughput but if removed I get full throughput.
    $redis->get('key');

    $response->end('Hello');
    $pool_redis->enqueue($redis);
});

$server->start();
```

swoole 4.2.10

**Comment by matyhtf** [comment]
Please use the coroutine client.
```php
$pool_redis = new \SplQueue;

$server = new swoole_http_server("127.0.0.1", 9501, SWOOLE_BASE);
$server->set(['worker_num' => 1]);

$server->on('request', function (swoole_http_request $req, swoole_http_response $response) use ($pool_redis) {
    if ($pool_redis->isEmpty()) {
        $redis = new swoole_redis;
        $redis->connect('127.0.0.1', 6379);
        $redis->set('key', 'A');
    } else {
        $redis = $pool_redis->dequeue();
    }
    //with this I get half throughput but if removed I get full throughput.
    $redis->get('key');

    $response->end('Hello');
    $pool_redis->enqueue($redis);
});

$server->start();
```

**Comment by jobs-git** [comment]
Hi, I forgot to mention that I already tested w/ coroutine before hand and got similar result. If it helps throughout the whole test, $redis sits idle.

I tried talking to Redis guys about this issue and they were blaming it to `overhead of system calls and networking`, this seems to be interesting, but I dont think I have enough knowledge to mingle with that.

These are the results of `time` command

```bash
user: 20.11 
system: 11.24 <---- (more than half of the user work)
elapsed: 0:39.89

pagefaults: (0major+10013minor) <--- (swaping in virtual mem?)
```
